sycl/test/xocc_tests/sdaccel_ports/vision/edge_detection/edge_detection_with_webcam.cpp

```diff
@@ -25,71 +25,68 @@ int main(int argc, char* argv[]) {
   constexpr auto area = height * width;
 
   cv::VideoCapture cap;
-  // set buffer to 1 frame
-  cap.set(cv::CAP_PROP_BUFFERSIZE, 1);
-  if(!cap.open(0))
+  // If opencv don't find a webcam the program exit here
+  if (!cap.open(0)) {
     return 0;
+  }
 
   // set the size of picture taken by the webcam
   cap.set(cv::CAP_PROP_FRAME_WIDTH, width);
   cap.set(cv::CAP_PROP_FRAME_HEIGHT, height);
 
   selector_defines::CompiledForDeviceSelector selector;
-  queue q {selector, property::queue::enable_profiling()};
+  queue q { selector, property::queue::enable_profiling {} };
 
-  // This infinite loop take a picture from the camera
-  // apply the edge detection filter with the FPGA kernel
-  // display the input and the output in separated windows
-  for(;;){
+  // This infinite loop takes a picture from the camera
+  // applies the edge detection filter with the FPGA kernel
+  // and displays the input and the output in separated windows
+  for (;;) {
     cv::Mat inputColor;
-    // take the picture frim camera
+    // take the picture from the camera
     cap >> inputColor;
 
     cv::imshow("inputColor", inputColor);
     cv::Mat inputRaw, input;
-    cv::cvtColor(inputColor, inputRaw, CV_BGR2GRAY);
     // convert the colored picture into grey
+    cv::cvtColor(inputColor, inputRaw, CV_BGR2GRAY);
     inputRaw.convertTo(input, CV_8UC1);
 
     buffer<uchar> ib{input.begin<uchar>(), input.end<uchar>()};
     buffer<uchar> ob{range<1>{area}};
 
     // The loop break when pressing the Esc key
-    if( cv::waitKey(10) == 27 ) break;
-
-    cv::imshow("input", input);
+    if (cv::waitKey(10) == 27) break;
 
     auto event = q.submit([&](handler &cgh) {
       auto pixel_rb = ib.get_access<access::mode::read>(cgh);
       auto pixel_wb = ob.get_access<access::mode::write>(cgh);
 
       cgh.single_task<xilinx::reqd_work_group_size<1, 1, 1, krnl_sobel>>([=] {
-        auto gX = xilinx::partition_array
-          <char, 9,xilinx::partition::complete<0>>
-          ({-1, 0, 1, -2, 0, 2, -1, 0, 1});
-
-        auto gY = xilinx::partition_array
-          <char, 9,xilinx::partition::complete<0>>
-          ({1, 2, 1, 0, 0, 0, -1, -2, -1});
+        xilinx::partition_array
+          <char, 9,xilinx::partition::complete<0>> gX
+          { {-1, 0, 1, -2, 0, 2, -1, 0, 1} };
+        xilinx::partition_array
+          <char, 9,xilinx::partition::complete<0>> gY
+          { {1, 2, 1, 0, 0, 0, -1, -2, -1} };
 
       int magX, magY, gI, pIndex;
       for (size_t x = 1; x < width - 1; ++x) {
         for (size_t y = 1; y < height - 1; ++y) {
-          magX = 0; magY = 0;
+          int magX = 0; magY = 0;
```

Comment:
What is this half-baked change?

Reply:
What magic do you use to have it to compile?
I am not even talking about execution... :-(

Reply:
It should have to be `int magX = 0, magY = 0;` in C++ I believe. `;` is a command termination character, so it's somewhat equivalent to if you had it on two separate lines with a termination character ending each. e.g.:

```
int magX = 0;
magY = 0;
```
So the second has no type right now.

The comma operator (which I think is what using `,` falls under in this case, but Ronan can correct me if I'm wrong) is just a way to sequence commands and has some specific semantics to it: 
https://en.cppreference.com/w/cpp/language/operator_other

Also not the same as using a comma in an initialization list or function call and one caveat is the expressions type is equivalent to the type of the final expression!

Hopefully somewhat helpful.